src/Stratis.Bitcoin.Features.BlockStore/ProvenHeadersBlockStoreBehavior.cs

```diff
@@ -59,11 +59,14 @@ protected override Payload BuildAnnouncedHeaderPayload(int blockstoreTipHeight,
         {
             if (this.AnnounceProvenHeadersFromHeight > 0 && blockstoreTipHeight >= this.AnnounceProvenHeadersFromHeight)
             {
-                return new ProvenHeadersPayload(
-                    from header in headers
-                    let posBlock = new PosBlock(header)
-                    select ((PosConsensusFactory)this.network.Consensus.ConsensusFactory).CreateProvenBlockHeader(posBlock)
-                    );
+                ProvenHeadersPayload provenHeadersPayload = new ProvenHeadersPayload();
+                foreach (var header in headers)
+                {
+                    var posBock = new PosBlock(header);
```

Comment:
This is not correct, you need to use `Network.ConsensusFctory.CreateBlock()`, fix it in the next PR

Reply:
CreateBlock doesn't have a header argument, should we add it?

Reply:
Ah hang on this is probably not correct, you don't just create the PH like this, the block you created is missing infomration, you need the block itself
but also potentially you will have the PH available in the PH store so I guess this is the next PR